common/models/user.js

```diff
@@ -993,18 +995,22 @@ module.exports = function(User) {
 
   User.validatePassword = function(plain) {
     var err;
-    if (plain && typeof plain === 'string' && plain.length <= MAX_PASSWORD_LENGTH) {
-      return true;
+    if (!plain || typeof plain !== 'string') {
+      err = new Error(g.f('Invalid password.'));
+      err.code = 'INVALID_PASSWORD';
```

Comment:
@loay thank you for the comment. I am afraid I'm with @kjdelisle here, but let me explain in more detail:

In my understanding, `validatePassword` is called only when setting/updating the password value, which something performed only by the user themselves, and therefore it's ok to tell them they provided an invalid password value.

> but for resetting passwords as well, so we could be pointing a potential hackers to where they should dig deeper

Assuming you mean "reset password" functionality - the new password is entered only after the person opened the link sent by email from the initial "password reset" page. In order to allow the attacker to access the page for entering a new password, they must have been able to read that password-reset email, in which case we have much bigger trouble than letting the attacker know the maximum password length!